Compilation failure when building without MPI

Compilation without MPI fails in accumulators.

```
[ 33%] Building CXX object accumulators/CMakeFiles/alps-accumulators.dir/src/feature/mean.cpp.o
In file included from ALPSCore/accumulators/src/feature/mean.cpp:12:
In file included from ALPSCore/accumulators/include/alps/accumulators/feature/mean.hpp:13:
In file included from ALPSCore/accumulators/include/alps/accumulators/feature/count.hpp:13:
ALPSCore/hdf5/include/alps/hdf5/archive.hpp:313:19: error: no member named 'save' in
      'std::vector<float, std::allocator<float> >'
            value.save(ar);
            ~~~~~ ^
ALPSCore/hdf5/include/alps/hdf5/archive.hpp:407:13: note: in instantiation of function template
      specialization 'alps::hdf5::save<std::vector<float, std::allocator<float> > >' requested here
            save(ar, proxy.path_, proxy.value_);
            ^
ALPSCore/hdf5/include/alps/hdf5/archive.hpp:440:21: note: in instantiation of function template
      specialization 'alps::hdf5::operator<<<const std::vector<float, std::allocator<float> > &>' requested here
                ar_ << make_pvp(path_, value);
                    ^
ALPSCore/accumulators/src/feature/mean.cpp:47:34: note: in instantiation of function template
      specialization 'alps::hdf5::detail::archive_proxy<alps::hdf5::archive>::operator=<std::vector<float,
      std::allocator<float> > >' requested here
                ar["mean/value"] = mean();
                                 ^
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -21,6 +21,10 @@ addons:
 
 install: true
 
+env:
+  - ENABLE_MPI=ON ALPS_CC=mpicc ALPS_CXX=mpic++
+  - ENABLE_MPI=OFF
+
 before_script:
   - export OMPI_CC=${CC}
   - export OMPI_CXX=${CXX}
@@ -36,10 +40,11 @@ script:
   - |
     cmake ..                                              \
     -DCMAKE_BUILD_TYPE=Debug                              \
-    -DCMAKE_C_COMPILER=mpicc                              \
-    -DCMAKE_CXX_COMPILER=mpic++                           \
+    -DCMAKE_C_COMPILER=${ALPS_CC:-${CC}}                  \
+    -DCMAKE_CXX_COMPILER=${ALPS_CXX:-${CXX}}              \
     -DCMAKE_INSTALL_PREFIX=$TRAVIS_BUILD_DIR/installed    \
-    -DALPS_INSTALL_EIGEN=true
+    -DALPS_INSTALL_EIGEN=true                             \
+    -DENABLE_MPI=$ENABLE_MPI
   - make -j3
   - make test
   - make install

--- a/accumulators/include/alps/accumulators/feature.hpp
+++ b/accumulators/include/alps/accumulators/feature.hpp
@@ -20,6 +20,7 @@
 #include <type_traits>
 
 #ifdef ALPS_HAVE_MPI
+    #include <alps/hdf5/archive.hpp>
     #include <alps/accumulators/mpi.hpp>
 #endif
 
@@ -219,4 +220,3 @@ namespace alps {
         }
     }
 }
-

--- a/accumulators/include/alps/accumulators/mpi.hpp
+++ b/accumulators/include/alps/accumulators/mpi.hpp
@@ -10,8 +10,6 @@
 
 #ifdef ALPS_HAVE_MPI
 
-    #include <alps/hdf5.hpp>
-
     #include <alps/utilities/boost_mpi.hpp>
 
     namespace alps {
@@ -24,4 +22,3 @@
     } // alps::
 
 #endif
-

--- a/accumulators/src/feature/binning_analysis.cpp
+++ b/accumulators/src/feature/binning_analysis.cpp
@@ -6,6 +6,8 @@
 
 #include <alps/accumulators/feature/error.hpp>
 #include <alps/accumulators/feature/binning_analysis.hpp>
+#include <alps/hdf5/vector.hpp>
+
 
 #include <boost/preprocessor/tuple/to_seq.hpp>
 #include <boost/preprocessor/seq/for_each.hpp>

--- a/accumulators/src/feature/error.cpp
+++ b/accumulators/src/feature/error.cpp
@@ -10,6 +10,7 @@
 #include <alps/config.hpp>
 
 #include <alps/accumulators/feature/error.hpp>
+#include <alps/hdf5/vector.hpp>
 
 #define ALPS_ACCUMULATOR_VALUE_TYPES_SEQ BOOST_PP_TUPLE_TO_SEQ(ALPS_ACCUMULATOR_VALUE_TYPES_SIZE, (ALPS_ACCUMULATOR_VALUE_TYPES))
 

--- a/accumulators/src/feature/max_num_binning.cpp
+++ b/accumulators/src/feature/max_num_binning.cpp
@@ -6,6 +6,7 @@
 
 #include <alps/accumulators/feature/max_num_binning.hpp>
 #include <alps/accumulators/feature/binning_analysis.hpp>
+#include <alps/hdf5/vector.hpp>
 
 #include <boost/preprocessor/tuple/to_seq.hpp>
 #include <boost/preprocessor/seq/for_each.hpp>

--- a/accumulators/src/feature/mean.cpp
+++ b/accumulators/src/feature/mean.cpp
@@ -10,6 +10,7 @@
 #include <alps/config.hpp>
 
 #include <alps/accumulators/feature/mean.hpp>
+#include <alps/hdf5/vector.hpp>
 
 #define ALPS_ACCUMULATOR_VALUE_TYPES_SEQ BOOST_PP_TUPLE_TO_SEQ(ALPS_ACCUMULATOR_VALUE_TYPES_SIZE, (ALPS_ACCUMULATOR_VALUE_TYPES))
 

--- a/accumulators/src/mpi.cpp
+++ b/accumulators/src/mpi.cpp
@@ -5,6 +5,7 @@
  */
 
 #include <alps/accumulators/mpi.hpp>
+#include <alps/hdf5.hpp>
 
 #ifdef ALPS_HAVE_MPI
 

